calico_node/startup/startup.go

```diff
@@ -771,6 +771,11 @@ func ensureDefaultConfig(c *client.Client, node *api.Node) error {
 		return err
 	}
 
+	// Update the ClusterType with the type in CLUSTER_TYPE
+	if err := ensureClusterTypeSet(c, "ClusterType", os.Getenv("CLUSTER_TYPE")); err != nil {
+		return err
+	}
+
```

Comment:
You can invoke this method for KDD, however - KDD does not yet support per-node configuration so you the call to `ensureFelixConfig` will fail.